src/main/java/backend/Logic.java

```diff
@@ -124,10 +126,17 @@ public void refresh(boolean isNotificationPaneShowing) {
             } else {
                 logger.info("Opening " + repoId);
                 UI.status.displayMessage("Opening " + repoId);
+                Platform.runLater(() -> UI.events.triggerEvent(new RepoOpeningEvent(repoId, isPrimaryRepository)));
```

Comment:
The `else` clause is not needed here. Not your code in this PR but would be nice if you help refactor this part :-P.